packages/flutter/lib/src/cupertino/route.dart

```diff
@@ -652,14 +647,14 @@ class _CupertinoBackGestureController<T> {
     }
     _animating = false;
     if (status == AnimationStatus.dismissed)
-      navigator.pop<T>(); // this will cause the route to get disposed, which will dispose us
+      route.navigator.removeRoute(route); // this will cause the route to get disposed, which will dispose us
```

Comment:
Turns out this has a subtle (untested :()problem. This will dispose the route first which means by the time the controller is disposed, the route's reference to the navigator is gone already and we can't call didStopUserGesture. Breaks a few uncommon APIs on the NavigatorObserver. I'll fix it in a next pr.

Reply:
Of course, I don’t understand how it all works, but this change caused this problem #29596, I added
route.navigator.pop ();
instead
route.navigator.removeRoute (route);
At first glance, everything works, but what this leads to, I cannot say) you know better

Reply:
Thanks so much for digging through this @1AlexFix1, I think you're right in that this would break existing NavigatorObserver.didPop. 

@HansMuller, let's chat about this next week. Though I'm not sure what is the right solution. Since didPop() actually returns the route that was popped whereas pop() doesn't let you specify which route was popped, this API asymmetry might be the root of the issue.